downsample_model.py

```diff
@@ -0,0 +1,242 @@
+
+
+def count_parameters(model):
+    """
+    Copied from: https://discuss.pytorch.org/t/how-do-i-check-the-number-of-parameters-of-a-model/4325/9
+    """
+    return sum(p.numel() for p in model.parameters() if p.requires_grad)
+
+def select_layers_from_strategy(strategy, n_layers, downsampling_rate):
+    """
+        Given the downsampling strategy, the number of layers and the downsampling rate
+        Return a dictionary that maps the layers we want to select from the source model 
+        to the layers we want to assign on the shriked model, in the format
+        {source_model_key:target_model_key, ... }
+        Note that this is applied only on the attention block layers (named as `h` in `BloomModel`)
+
+        Example:
+            downsampling rate = 0.5 (ie consider half of the layers)
+            Source Model: 
+                h.0 | h.1 | ... | h.30
+            Target Model:
+                if strategy = 'first':
+                    returns {h.0:h.0, h.1:h.1, ... h.14:h.14} (consider half of the layers)
+                if strategy = 'last':
+                    returns {h.15:h.0, h.16:h.0, ... , h.30:h.14}
+                    So h.15 of the source model will correspond to h.0 in the target model, etc.
+                if strategy = 'step':
+                    returns {h.0:h.0, h.2:h.1, h.4:h.0, ... , h.30:h.14}
+                    We consider the layers of the source model with a step of 1/downsample_rate
+        Args:
+            strategy (`str`, *required*):
+                layer selection strategy, must be `first`, `last` or `step` (see comments above)
+            n_layers (`int`, *required*):
+                total number of layers in the source model
+            downsampling_rate (`float`, *required*):
+                width downsampling rate
+    """
+    selecting_rate = int(1/(downsampling_rate))
+
+    if strategy == "first":
+        array_layers = ["h."+str(i) for i in range(selecting_rate)]
+    elif strategy == "last":
+        array_layers = ["h."+str(i) for i in range(int(selecting_rate), n_layers)]
+    elif strategy == "step":
+        array_layers = ["h."+str(i) for i in range(n_layers) if i % int(selecting_rate) == 0]
+    else:
+        raise NotImplementedError("Unknown strategy: {}".format(strategy))
+    return {layer:"h."+str(i) for i, layer in enumerate(array_layers)}
+
+def convert_config_to_downsized_config(config, hidden_size_downsampling_rate, width_downsampling_rate, aggregation_strategy, layer_selection_strategy):
+    """
+        Defines the new hyperparameters of the shrinked model, given the downsampling rates.
+        Creates a new config file and returns it.
+        Args:
+            config (`transformers.BloomConfig`, *required*):
+                input old model's config
+            hidden_downsampling_rate (`float`, *required*):
+                hidden states downsampling rate
+            width_downsampling_rate (`float`, *required*):
```

Comment:
I think you mean `depth_downsampling_rate`

Reply:
yeah true! thanks for the nit
FIxed in 854dccf92fee29818a6cfc69e6ebc6e549a6ac7f